### PR TITLE
Dry-run mode

### LINKED ISSUE #7: Add dry-run option
Add a dry-run option to the command to let user dry-run the workflow to make sure their workflow is valid before running it in the actual environment.

### PR DESCRIPTION
**Pull request by jsidorenko**
Closes #7

### CODE REVIEW

**Comment by hamidra** [comment]
Some other stuff that we can consider as part of dryRun:
-Can you also check if we can use dryRun [RPC](https://polkadot.js.org/docs/substrate/rpc) to dryRun the chain txs as well when that RPC exists by the chain, .
api.rpc.system.dryRun

-We can verify when a metadata is specified the corresponding media files are available in the instance.metadata.imageFolder per each minting row (e.x. if it mints rows from 1-10 and the image.metadat.extension is png  1.png ... 10.png should be available in the specified directory)

File: src/index.js [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
 program
   .parseAsync(process.argv)
   .then(() => process.exit(0))
   .catch((err) => {
-    if (err instanceof WorkflowError) {
-      console.error(err?.message);
-    } else {
-      console.error(err);
-    }
+    console.error(err?.message ?? err);
```
Comment (hamidra) [inline src/index.js]: This is nice, but I was thinking for any error that is not related to the workflow and is not a user error, we want to print the whole trace for debugging?! generally this should not happen unless we have a bug or mishandling the errors.

File: src/workflow/balanceTransfer.js [hunk 2]
```diff
@@ -12,8 +12,9 @@ let transferFunds = async (network, owners, amount) => {
   let call = proxiedAddress
     ? api.tx.proxy.proxy(proxiedAddress, 'Assets', txBatch)
     : txBatch;
-  await signAndSendTx(api, call, signingPair);
-  console.log(call.toHuman());
+  await signAndSendTx(api, call, signingPair, true, dryRun);
+
+  if (!dryRun) console.log(call.toHuman());
```
Comment (hamidra) [inline src/workflow/balanceTransfer.js]: This is probably some leftover from my debuggings. can be deleted.

File: src/workflow/metadata.js [hunk 3]
```diff
@@ -58,35 +60,45 @@ let setMetadataInBatch = async (connection, classId, instanceMetaCids) => {
   let call = proxiedAddress
     ? api.tx.proxy.proxy(proxiedAddress, 'Assets', txBatch)
     : txBatch;
-  await signAndSendTx(api, call, signingPair);
-  console.log(call.toHuman());
+  await signAndSendTx(api, call, signingPair, true, dryRun);
+
+  if (!dryRun) console.log(call.toHuman());
```
Comment (hamidra) [inline src/workflow/metadata.js]: This can be deleted.

File: src/workflow/metadata.js [hunk 4]
```diff
@@ -58,35 +60,45 @@ let setMetadataInBatch = async (connection, classId, instanceMetaCids) => {
...
+      pinataClient,
+      name,
+      description,
+      imageFile
+    ));
+  } else {
+    metaCid = randomAsHex(23).replace('0x', '');
+  }
+
+  await setClassMetadata(connection, classId, metaCid, dryRun);
```
Comment (hamidra) [inline src/workflow/metadata.js]: Shouldn't this be reverted since we are not doing tx dryRun anymore?!

**Review by hamidra — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (jsidorenko) [reply src/index.js]: Got it, I'll revert it then

File: src/workflow/metadata.js (hunk 4 above)
Comment (jsidorenko) [reply src/workflow/metadata.js]: Will do

**Review by hamidra — APPROVED** [review]

**Pull request merged**